chapters/arrays.tex

```diff
@@ -195,10 +195,14 @@ \section{Built-in Array Functions}\doublelabel{built-in-array-functions}
 \lstinline[mathescape=true]!size(C, $j$)! = $1$ for $n_{\mathrm{A}} + 1 \leq j \leq n$, \lstinline[mathescape=true]!C[$i_{1}$, $\ldots$, $i_{n_{\mathrm{A}}}$, 1, $\ldots$, 1]! =
 \lstinline[mathescape=true]!A[$i_{1}$, $\ldots$, $i_{n_{\mathrm{A}}}$]!\\ \hline
 \end{longtable}
+The argument $n$ must be literal or a constant that can be evaluated during translation because it determines
+the number of dimensions of the returned array.
 
 \begin{nonnormative}
-The function \lstinline!promote! cannot be used in Modelica, because the number of dimensions of the returned array cannot be determined at compile time
-if $n$ is a variable.  Below, \lstinline!promote! is only used for constant $n$.
+Below, \lstinline!promote! is only used for constant $n$.
+
+Varying $n$ for \lstinline!promote! complicates matrix handling as it
+can change matrix-equations in subtle ways (e.g.\ changing inner products to matrix multiplication).
```

Comment:
I had thought about adding some text about this function usually not being needed. But I guess it's superfluous information and easier to just add promote as it is.